google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/BigtableBatchingCallSettings.java

```diff
@@ -145,9 +194,137 @@ public Builder setRetrySettings(@Nonnull RetrySettings retrySettings) {
+      this.isLatencyBasedThrottlingEnabled = true;
+      this.targetRpcLatencyMs = targetRpcLatency;
+      return this;
+    }
+
+    /** Disable latency based throttling. */
+    public Builder disableLatencyBasedThrottling() {
+      this.isLatencyBasedThrottlingEnabled = false;
+      this.targetRpcLatencyMs = null;
+      return this;
+    }
+
+    /** Gets target rpc latency if latency based throttling is enabled. Otherwise returns null. */
+    @Nullable
+    public Long getTargetRpcLatencyMs() {
+      return isLatencyBasedThrottlingEnabled ? targetRpcLatencyMs : null;
+    }
+
+    /** Gets if latency based throttling is enabled. */
+    public boolean isLatencyBasedThrottlingEnabled() {
+      return this.isLatencyBasedThrottlingEnabled;
+    }
+
+    /**
+     * Gets the {@link DynamicFlowControlSettings} that'll be used to set up a {@link
+     * FlowController} for throttling.
+     *
+     * <p>By default, this will allow a maximum of 1000 entries per channel of {@link
+     * FlowControlSettings.Builder#setMaxOutstandingElementCount request count} and 100MB of {@link
+     * FlowControlSettings.Builder#setMaxOutstandingRequestBytes accumulated size} in-flight
+     * requests. Once the limits are reached, pending operations will by default be {@link
+     * FlowControlSettings.Builder#setLimitExceededBehavior blocked} until some of the in-flight
+     * requests are resolved.
+     *
+     * <p>If latency based throttling is enabled, number of entries allowed by {@link
+     * FlowController} will be adjusted to reach {@link Builder#getTargetRpcLatencyMs()}.
+     *
+     * <ul>
+     *   <li>{@link FlowController} will be set to allow Math.max({@link BatchingSettings.Builder
+     *       #setElementCountThreshold batch size}, {@link
+     *       FlowControlSettings.Builder#setMaxOutstandingElementCount request count} / 4) entries
+     *       to start with.
+     *   <li>If bulk mutation rpc latency is higher than target latency, decrease allowed entries to
+     *       a minimum of Math.max({@link BatchingSettings.Builder#setElementCountThreshold batch
+     *       size}, {@link FlowControlSettings.Builder#setMaxOutstandingElementCount request count}
+     *       / 100).
+     *   <li>If bulk mutation rpc latency is lower than target latency and there was throttling,
+     *       increase allowed entries to a maximum of {@link
+     *       FlowControlSettings.Builder#setMaxOutstandingElementCount request count}.
+     * </ul>
+     *
+     * If latency based throttling is disabled, {@link FlowController} will always allow {@link
+     * FlowControlSettings.Builder#setMaxOutstandingElementCount request count}.
+     *
+     * <p>Latency based throttling only updates outstanding entries count. {@link FlowController}
+     * will always allow {@link FlowControlSettings.Builder#setMaxOutstandingRequestBytes
+     * accumulated size}.
+     */
+    public DynamicFlowControlSettings getDynamicFlowControlSettings() {
```

Comment:
Does this need to be public? Its a bit assymmetric with it being package protected above